ui/app/pages/swaps/view-quote/view-quote-price-difference.js

```diff
@@ -0,0 +1,109 @@
+import React, { useContext } from 'react'
+
+import PropTypes from 'prop-types'
+import classnames from 'classnames'
+import BigNumber from 'bignumber.js'
+import { useEthFiatAmount } from '../../../hooks/useEthFiatAmount'
+import { I18nContext } from '../../../contexts/i18n'
+
+import ActionableMessage from '../actionable-message'
+import Tooltip from '../../../components/ui/tooltip'
+
+export default function ViewQuotePriceDifference(props) {
+  const t = useContext(I18nContext)
+
+  const { usedQuote, sourceTokenValue, destinationTokenValue } = props
+
+  const priceSlippageFromSource = useEthFiatAmount(
+    usedQuote?.priceSlippage?.sourceAmountInETH || 0,
+  )
+  const priceSlippageFromDestination = useEthFiatAmount(
+    usedQuote?.priceSlippage?.destinationAmountInEth || 0,
+  )
+  const priceSlippageUnknownFiatValue =
+    !priceSlippageFromSource || !priceSlippageFromDestination
```

Comment:
Is this the correct way to detect if we should show the unknown fiat message?

Reply:
this, or if there is a calculation error